google/cloud/pubsublite/internal/wire/single_partition_publisher.py

```diff
@@ -120,11 +135,25 @@ def _handle_response(self, response: PublishResponse):
                     "Received an publish response on the stream with no outstanding publishes."
                 )
             )
-        next_offset: Cursor = response.message_response.start_cursor.offset
-        batch: List[WorkItem[PubSubMessage]] = self._outstanding_writes.pop(0)
-        for item in batch:
-            item.response_future.set_result(Cursor(offset=next_offset))
-            next_offset += 1
+        ranges = response.message_response.cursor_ranges
+        ranges.sort(key=_get_start_index)
+        batch: List[WorkItem[_SequencedMessage]] = self._outstanding_writes.pop(0)
+        rangeIdx = 0
+        for msgIdx, item in enumerate(batch):
+            if rangeIdx < len(ranges) and ranges[rangeIdx].end_index <= msgIdx:
+                rangeIdx += 1
+            offset = -1
+            if (
+                rangeIdx < len(ranges)
+                and msgIdx >= ranges[rangeIdx].start_index
+                and msgIdx < ranges[rangeIdx].end_index
+            ):
+                offset = (
+                    (ranges[rangeIdx].start_cursor.offset or 0)
+                    + msgIdx
+                    - ranges[rangeIdx].start_index
```

Comment:
nit. give this a name? 

offset_in_range = msg_id - ranges[rangeIdx].start_index

Reply:
Done